samples/nrf_desktop/src/hw_interface/buttons.c

```diff
@@ -335,19 +366,31 @@ void button_pressed(struct device *gpio_dev, struct gpio_callback *cb,
 static void init_fn(void)
 {
 	/* Setup GPIO configuration */
-	gpio_dev = device_get_binding(DT_GPIO_P0_DEV_NAME);
-	if (!gpio_dev) {
-		LOG_ERR("Cannot get GPIO device binding");
-		return;
+	const char *gpio_dev_names[] = {DT_GPIO_P0_DEV_NAME,
+					DT_GPIO_P1_DEV_NAME};
+
+	for (size_t i = 0; i < GPIO_PORT_NUM; i++) {
+		if (port_used(i)) {
+			gpio_devs[i] = device_get_binding(gpio_dev_names[i]);
+			if (!gpio_devs[i]) {
+				LOG_ERR("Cannot get GPIO device binding");
+				return;
+			}
+		}
 	}
 
-	for (size_t i = 0; i < ARRAY_SIZE(col_pin); i++) {
-		int err = gpio_pin_configure(gpio_dev, col_pin[i],
-				GPIO_DIR_OUT);
+	if (IS_MATRIX) {
```

Comment:
there is no need for this check - for loop will do no cycles as `ARRAY_SIZE(col)` in such case is zero.